pkg/webhook/parser.go

```diff
@@ -228,15 +238,15 @@ func (c Config) clientConfig() admissionreg.WebhookClientConfig {
 }
 
 // sideEffects returns the sideEffects config for a webhook.
-func (c Config) sideEffects() *admissionreg.SideEffectClass {
-	var sideEffects admissionreg.SideEffectClass
+func (c Config) sideEffects() *admissionregv1.SideEffectClass {
+	var sideEffects admissionregv1.SideEffectClass
 	switch strings.ToLower(c.SideEffects) {
-	case strings.ToLower(string(admissionreg.SideEffectClassNone)):
-		sideEffects = admissionreg.SideEffectClassNone
-	case strings.ToLower(string(admissionreg.SideEffectClassNoneOnDryRun)):
-		sideEffects = admissionreg.SideEffectClassNoneOnDryRun
-	case strings.ToLower(string(admissionreg.SideEffectClassSome)):
-		sideEffects = admissionreg.SideEffectClassSome
+	case strings.ToLower(string(admissionregv1.SideEffectClassNone)):
+		sideEffects = admissionregv1.SideEffectClassNone
+	case strings.ToLower(string(admissionregv1.SideEffectClassNoneOnDryRun)):
+		sideEffects = admissionregv1.SideEffectClassNoneOnDryRun
+	case strings.ToLower(string(admissionregv1.SideEffectClassSome)):
+		sideEffects = admissionregv1.SideEffectClassSome
 	case "":
 		return nil
 	default:
```

Comment:
Nit: combine the `””` and default cases into one default that returns `None`, since this is a required value for v1 configs. Then you can remove this for v1beta1 in the same loop with `AdmissionReviewVersions` below.

Reply:
I don't think this will work, because the default value for `v1beta1` is `Unknown` if we default it to `None` here, in the loop below, we can not distinguish if user set it to `None` or the user didn't set it.

Reply:
I added a check in the for loop to default it to `None` for `v1`